lib/ReactViews/Preview/measureWidth.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import debounce from 'lodash.debounce';
+
+const getDisplayName = WrappedComponent => {
+    return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+};
+
+/*
+    HOC to check component width & supply updated widths as a prop (stored interally via state)
+    Ensure the wrapped component contains a reference to the element you wish to measure the width of,
+    typically the root element. Do this via a callback instead of `React.createRef()` as it was written for pre-react-16.3
```

Comment:
Sounds like this originally came from another project. Are we definitely ok to use it?

Reply:
I originally wrote it for another project and tied it in quite heavily to redux - so it's been adapted to what we have here, but I will confirm the definitely part & report back - in the meantime I may just make it work with both styles of refs as it's not like the callback ref has been deprecated.

Reply:
I'm not too fussed about the callback versus createRef, just wanted to make sure there are no IP issues.

Reply:
Yes hence the confirming the definitely part :) Should get an answer tomorrow.

Reply:
"of course!"
No issues here!